src/handlers.js

```diff
@@ -98,22 +82,23 @@ const createReportHandlers = (reportsConfig) => {
                     // value. We'll treat these as null, rather than an empty string.
                     ...fromEntries(Array.from(ctx.request.URL.searchParams.entries()).filter(([, v]) => v !== '')),
                 };
-                ctx.state.logger.push({ dbQuery, queryArgs }).log('Executing query');
-                const result = await ctx.db.query(dbQuery, queryArgs);
-                ctx.state.logger.push({ dbQuery, queryArgs, result }).log('Query result');
 
-                ctx.response.body = result;
+                const queries = Object.entries(template.dataSource.data)
+                    .map(([k, v]) => ctx.db.query(v, queryArgs)
+                        .then((result) => ({ [k]: result })));
+                const result = await Promise.all(queries);
+                ctx.response.html = template.render(Object.assign({}, ...result));
             },
         };
-        return [route, handler];
+        return [`/${route}`, handler];
     };
 
-    return fromEntries(Object.entries(reportsConfig).map(createReportHandler));
+    return fromEntries(Object.entries(reportTemplates).map(createReportHandler));
```

Comment:
`fromEntries` is probably actually available now as `Object.fromEntries`. I guess the shim will use that if it's available, so no huge rush to get rid of it if you can't be bothered right now.

Reply:
done in PR https://github.com/mojaloop/reporting/pull/17